lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/client/api/JobSubmissionEventImpl.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.common.client.api;
+
+import org.apache.reef.runtime.common.files.FileResource;
+import org.apache.reef.util.BuilderUtils;
+import org.apache.reef.util.Optional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class JobSubmissionEventImpl implements JobSubmissionEvent {
+  private final String identifier;
+  private final String remoteId;
+  private final String configuration;
+  private final String userName;
+  private final List<FileResource> globalFileList;
```

Comment:
Yes, that was the initial idea. But we don't make use of that anymore. The distinctions were:

  * File / Library: For now, we load all DLLs / add all JARs to the classpath. 
  * Archive / File: We don't do anything special with archives. Instead, we ZIP up the whole submission and upload it to HDFS.

Hence, I am not sure whether we should keep the distinction. The library / file distinction is somewhat useful. For that, we could transfer two sets of files.

Also, please use `Set<>` instead of `List<>`. We don't support the case when the same file is listed twice.